src/sov/mod.rs

```diff
@@ -0,0 +1,295 @@
+            let Ok(source_url) = settings.sov_source.parse::<url::Url>() else {
+                return Err(
+                    HandlerErrorKind::Internal(
+                        format!("Unable to parse SOV URL '{}'", &settings.sov_source)
+                    ).into()
+                );
+            };
+
+            return Ok(SOVManager {
+                source_url: Some(source_url),
+                refresh_rate: Duration::from_secs(settings.sov_refresh_rate_secs),
+                encoded_sov: None,
+                last_response: None,
+            });
+        }
+
+        // Then check if it's a local settings file or an inline settings string.
+        let sov = if Path::new(&settings.sov_source).exists() {
+            let Ok(sov) =
+                read_to_string(&settings.sov_source)
+                    .map_err(|_| Err::<String, &str>("Unable to read SOV settings file"))
+                    .and_then(|content| {
+                        serde_json::from_str::<SOVResponse>(&content)
+                            .map_err(|_| Err("Unable to load SOV settings from JSON"))
+                    }) else {
+                        return Err(
+                            HandlerErrorKind::Internal(
+                                format!("Unable to parse SOV settings from file '{}'", settings.sov_source)
+                            ).into()
+                        );
+                    };
+            sov
+        } else {
+            // Presume it's an inline SOV settings string.
+            let Ok(sov) = serde_json::from_str::<SOVResponse>(&settings.sov_source) else {
+                return Err(
+                    HandlerErrorKind::Internal(
+                        format!("Could not parse SOV settings inline: {:?}", &settings.sov_source)
+                    ).into()
+                );
+            };
+            sov
+        };
+
+        Ok(SOVManager {
+            source_url: None,
+            refresh_rate: Duration::from_secs(settings.sov_refresh_rate_secs),
+            encoded_sov: Some(
+                // Unwrapping is safe here since the parsing above ensures it's a valid `SOVResponse`.
+                base64::engine::general_purpose::STANDARD_NO_PAD
+                    .encode(serde_json::to_string(&sov).unwrap()),
+            ),
+            last_response: Some(LastResponse {
+                response: sov,
+                updated: Utc::now(),
+            }),
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use serde_json::json;
+
+    use super::*;
+    use crate::web::test::get_test_settings;
+
+    const MOCK_SOV: &str = "eyJuYW1lIjoiU09WLTIwMjMwNTE4MjE1MzE2IiwiYWxsb2NhdGlv\
+    bnMiOlt7InBvc2l0aW9uIjoxLCJhbGxvY2F0aW9uIjpbeyJwYXJ0bmV\
+    yIjoiYW1wIiwicGVyY2VudGFnZSI6MTAwfV19LHsicG9zaXRpb24iOj\
+    IsImFsbG9jYXRpb24iOlt7InBhcnRuZXIiOiJhbXAiLCJwZXJjZW50Y\
+    WdlIjo4OH0seyJwYXJ0bmVyIjoibW96LXNhbGVzIiwicGVyY2VudGFn\
+    ZSI6MTJ9XX1dfQ";
```

Comment:
Nit: for my OSD, sorry!
```suggestion
    const MOCK_SOV: &str = "eyJuYW1lIjoiU09WLTIwMjMwNTE4MjE1MzE2IiwiYWxsb2NhdGl\
                            vbnMiOlt7InBvc2l0aW9uIjoxLCJhbGxvY2F0aW9uIjpbeyJwYX\
                            J0bmVyIjoiYW1wIiwicGVyY2VudGFnZSI6MTAwfV19LHsicG9za\
                            XRpb24iOjIsImFsbG9jYXRpb24iOlt7InBhcnRuZXIiOiJhbXAi\
                            LCJwZXJjZW50YWdlIjo4OH0seyJwYXJ0bmVyIjoibW96LXNhbGV\
                            zIiwicGVyY2VudGFnZSI6MTJ9XX1dfQ";

```